apps/mobile/src/app/(auth)/sign-in.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from 'react';
+import { StyleSheet, Text, TextInput, View } from 'react-native';
+import { router } from 'expo-router';
+
+import { useAppDispatch, useAppSelector } from '@app/hooks/hooks';
+import type { SignInRequestDto } from '@app/types/types';
+import { selectIsAuthenticated, signIn } from '@app/store/auth/auth-slice';
+
+import { ScreenWrapper } from '@app/components/components';
+
+const SignInScreen = () => {
+	const dispatch = useAppDispatch();
+	const isAuthenticated = useAppSelector(state => selectIsAuthenticated(state));
+
+	const [user, setUser] = useState<SignInRequestDto>({
+		login: '',
+		password: '',
+	});
+
+	const handleSignIn = () => {
+		if (!user.login || !user.password) {
+			return;
+		}
+
+		dispatch(signIn(user));
+	};
+
+	useEffect(() => {
+		if (isAuthenticated) {
+			router.navigate('/(tabs)/home');
+		}
+	}, [isAuthenticated]);
+
+	return (
+		<ScreenWrapper>
+			<View style={styles.container}>
+				<Text>Sign In</Text>
+				<TextInput
+					placeholder="Email"
+					style={styles.input}
+					autoCapitalize="none"
+					onChangeText={text => setUser(prev => ({ ...prev, login: text }))}
```

Comment:
What do you think about creating a single function to handle state change? Something like 
const handleStateChange = (setEntityState) => (newState) => setEntityState(prev => ({...prev, ...newState})) 
We will be able to reuse it and avoid copy-pasting
@fshabanov what do you think?

Reply:
This needs to be handled with react-hook-form. We will not use local state to save form data